### ISSUE TITLE
Model backspin in `PhysicsBall`

### ISSUE THREAD

**Issue by jonathanlew** [UBC-Thunderbots/Software#1798]
### Description of the task
The current `PhysicsBall` model does not model any spin; it only models sliding and rolling friction. The `PhysicsBall` should be improved so that we can kick with some specified backspin and model the forces that act on the ball after it is kicked. Note that the dribbler currently applied a force to the middle of the dribbler rather than model any sort of spin on the ball.

### Acceptance criteria

- [ ] interface to apply backspin on `PhysicsBall`
- [ ] model `PhysicsBall` after it leaves the robot
- [ ] unit tests

### Blocked By
Mechanical team ball model

**Comment by jonathanlew** [comment]
no longer relevant given experimentation with other simulators
